### ISSUE TITLE
Kashmiri Apple Plant Disease Detection

### ISSUE THREAD

**Issue by abhisheks008** [World-of-ML/DL-Simplified#212]
### Deep Learning Simplified Repository (Proposing new issue)
:red_circle: **Project Title** : Kashmiri Apple Plant Disease Detection</br>
:red_circle: **Aim** : Create a DL model which will identify the Kashmiri Apple Plant Disease.</br>
:red_circle: **Dataset** : https://www.kaggle.com/datasets/hsmcaju/d-kap</br>
:red_circle: **Approach** : Try to use 3-4 algorithms to implement the models and compare all the algorithms to find out the best fitted algorithm for the model by checking the accuracy scores. Also do not forget to do a exploratory data analysis before creating any model.

**********************************************************************************
### 📍 **Follow the Guidelines to Contribute in the Project :**
* You need to create a separate folder named as the Project Title.
* Inside that folder, there will be four main components.
   * Images - To store the required images.
   * Dataset - To store the dataset or, information/source about the dataset.
   * Model - To store the machine learning model you've created using the dataset.
   * `requirements.txt` - This file will contain the required packages/libraries to run the project in other machines.
* Inside the `Model` folder, the `README.md` file must be filled up properly, with proper visualizations and conclusions.

***********************************************************************
:red_circle::yellow_circle: **Points to Note :**

- The issues will be assigned on a first come first serve basis, 1 Issue == 1 PR.
- "Issue Title" and "PR Title should be the same. Include issue number along with it.
- Follow Contributing Guidelines & Code of Conduct before start Contributing.

***********************************************************************
:white_check_mark: **To be Mentioned while taking the issue :**
- Full name : 
- GitHub Profile Link : 
- Email ID :
- Participant ID (if applicable):
- Approach for this Project :
- What is your participant role? (Mention the Open Source program)

*************************************************************
Happy Contributing 🚀 

All the best. Enjoy your open source journey ahead. 😎

**Comment by yashgosa** [comment]
Hey ! @abhisheks008 I would like to work on this project! 


# Introduction

Full name : Yash Chandrakant Gosavi
GitHub Profile Link : [yashgosa](https://github.com/yashgosa)
Email ID: yashcgosavi@gmail.com
Participant ID (if applicable):  *if there is, where can I find it ?*
What is your participant role? SSOC

## Description:

Farmers every year face economic loss and crop waste due to various diseases in Kashmiri apples. We will use image classification using CNN and build a mobile app using which a farmer can take a picture and the app will tell you if the plant has a disease or not.

## Technical Architecture of the Project

![image](https://github.com/World-of-ML/DL-Simplified/assets/113308290/2683e2e5-4a61-458d-b9ae-433d8279cb7b)

## Tech Stack used:

### 1. Model Building

1. Tensorflow
2. CNN
3. Data Augmentation
4. tf. dataset

### 2. Backend Server

1. tf. serving
2. FastAPI

### 3. Model Optimization

1. Quantization
2. tf. lite

### 4. Front End

1. React.js
2. React Native

### 5. Deployment (Maybe)

1. GCP

## Data Collection

We will be using this [Kashmiri Apple Plant Disease Dataset](https://www.kaggle.com/datasets/hsmcaju/d-kap) from Kaggle

**Comment by yashgosa** [comment]
@abhisheks008 is it possible to assign a mentor to me? So that if I get stuck somewhere s/he could help me

**Comment by abhisheks008** [comment]
Sure @yashgosa. It is a great to approach an issue. This issue will be assigned to you by June 1, once the program starts officially.

I will share the details of the mentors, you can connect them in the project channel.

**Comment by yashgosa** [comment]
Thanks @abhisheks008 ! I thought the program had already started 😅. Also where are you going to share the mentor details?

**Comment by abhisheks008** [comment]
No the program will start on June 1. A separate project channel will be created by the SSOC team in the discord, all the communications will be done there only.

**Comment by abhisheks008** [comment]
Issue assigned to @yashgosa

**Comment by Nithish-456** [comment]
Full name : Paidimarri Nithish
GitHub Profile Link : github.com/Nithish-456
Email ID : nithishpaidimarri@gmail.com
Participant ID (if applicable):
Approach for this Project : Using CNN or using transfer learning approach by pre trained models with early stopping, dropout regularization techniques  for classification of different diseases of apple plant. And building a streamlit GUI for easy user interaction for farmers to upload a photo and classify the particular disease associated with that plant. So, this project will useful for farmers effectively.
What is your participant role? (Mention the Open Source program): SWOC2024

**Comment by abhisheks008** [comment]
Try to use at least 2-3 deep learning methods for this project, compare them based on the accuracy scores to find out the best fitted model.

Issue assigned to you @Nithish-456

**Comment by Thewhitewolfsasi** [comment]
Hi, sir @abhisheks008 , I listed my approach down, please go through it and assign me under tag GSSOC'24 

Full name : Sasidharan V
GitHub Profile Link : https://github.com/Thewhitewolfsasi/
Email ID : sasistudy098@gmail.com
Participant ID (if applicable): 
Approach for this Project : 
Algorithms - CNN architecture model such VGG16, RESNET50 can be classify the images. 
Preprocessing - Image resizing, normalization, encoding and augmentation if required
Model Comparison and Selection - Evaluate Performance of all models based on the metrics obtained and will Choose the model that shows the best balance between accuracy, generalizability, and computational efficiency
What is your participant role? GSSOC'24

**Comment by abhisheks008** [comment]
Issue assigned to you @Thewhitewolfsasi

**Comment by binguliki** [comment]
Hi @abhisheks008  👋,

It has been three weeks since the project was assigned, and If there hasn't been significant progress on it. I'd like to propose taking up this issue to move things forward.

Here’s my planned approach:

1. **Data Augmentation**: Given the small size of the dataset, I'll start with some preprocessing to enhance it. This will include various augmentation techniques such as rotation, mirroring, zooming, and creating multiple combinations to increase the dataset's diversity.

2. **Model Implementation**:
    - I will initially implement the basic LeNet-5 architecture from scratch.
    - Additionally, I'll leverage some pretrained models such as InceptionResNet, VGG19, and AlexNet to compare performances.

3. **Hyperparameter Tuning**: To ensure optimal performance, I will conduct hyperparameter tuning on the best-performing model.

Could you please assign this task to me under GSSoC'24 with an appropriate level tag?

**Comment by binguliki** [comment]
@abhisheks008  Hey bro can you please check this .

**Comment by abhisheks008** [comment]
Already assigned to someone.

**Comment by vaishnavi003-svg** [comment]
please assign this issue to me.
thank you
